src/LearnositySdk/Request/Init.php

```diff
@@ -254,7 +289,12 @@ public function generateSignature()
 
         // Add the requestPacket if necessary
         if ($this->signRequestData && !empty($this->requestString)) {
-            array_push($signatureArray, $this->requestString);
+            // remove `meta` to not be a part of a signature hash
+            $requestStringDecoded = json_decode($this->requestString, true);
+            unset($requestStringDecoded['meta']);
+            $requestStringEncoded = json_encode($requestStringDecoded);
+
+            array_push($signatureArray, $requestStringEncoded);
```

Comment:
@shtrom @ceralena those lines make sure that `meta` is not being part of a signature hash, but ofc same changes should be done for LPs because otherwise our APIs won't work. I had to do it to fix tests.

**But this also produces a following issue:**

We want clients to be able to pass a request as a string to be signed, and this string should never be mutated, because it will be used to init API as well. We don't want it to be mutated as decoding and encoding causes the floating point numbers issue we've fixed a while ago. Us adding `meta` causes the mutation to happen, meaning that behind the scenes there is decoding and encoding happening.

For me it's a bit of a blocker for the whole telemetry... (even if floating point numbers issue is a very rare edge case). What do you think?

Reply:
Ah, reading the code, I see your point: We add the whole string of the request... This is fine on the client's serverside, but you are right that we run a risk on our serverside when generating the signature again....

Which APIs require the `request` data to be signed?

Reply:
Items, Reports, Author and Data.

This is why here I remove meta from signing, but this requires decoding the json string :/ I'm kind of stuck and can't think about a good solution to make this work, and not face that floating point bug... 

One could of course argue that this bug was so rare, that we shouldn't really care, but this would have to be properly reviewed, as the decision may have great implications.

@sebablanco I remember you mentioned a while ago another reason why we allowed passing requests as string, so it won't get mutated by decoding and encoding again. Could you remind me what was that case?

Reply:
Yes, that's my thought, too.

Reply:
👍 I may have mixed something up but agree with above comments, and @shtrom heavily pushing back on those changes ;)

Reply:
“Heavily”? It's feather-light butterfly touch q: